bokehjs/src/coffee/document.coffee

```diff
@@ -23,7 +24,10 @@ export class ModelChangedEvent extends DocumentChangedEvent
       throw new Error("'id' field should never change, whatever code just set it is wrong")
 
     value = @new_
-    value_json = HasProps._value_to_json('new_', value, @model)
+    if @model instanceof HasProps
+      value_json = @model.constructor._value_to_json(@attr, value, @model)
+    else
+      value_json = HasProps._value_to_json(@attr, value, @model)
```

Comment:
Yes, this is a bit awkward and needless. Is there an issue calling ``@model.constructor._value_to_json`` instead of encapsulating it all on HasProps. The ``ColumnDataSource`` overrides the ``_value_to_json`` static method to perform the serialization. Other suggestions on where this serialization should take place also welcome.